docs/source/howto/exploring.rst

```diff
@@ -0,0 +1,137 @@
+
+ * ``node_class``: accepts a subclass of :class:`~aiida.orm.nodes.node.Node`, only neighboring nodes with a class that matches this will be returned
+ * ``link_type``: accepts a vale of :class:`~aiida.common.links.LinkType`, only neighboring nodes that are linked with this link type will be returned
+ * ``link_label_filter``: accepts a string regex (with optional wildcards), only neighboring nodes that are linked with a link label that matches the regex will be returned
+
+As an example:
+
+.. code-block:: python
+
+    node.get_incoming(node_class=Data, link_type=LinkType.INPUT_CALC, link_label_filter='output%node_').all_nodes()
+
+will return only neighboring data nodes that are linked to the ``node`` with a link of type ``LinkType.INPUT_CALC`` and where the link label matches the regex ``'output%node_'``.
+The ``%`` character matches one or more characters and the ``_`` character matches exactly one character.
+
+
+.. _how-to:exploring:inputs-outputs:
+
+Inputs and outputs
+==================
+
+The :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing` methods, described in the :ref:`previous section <how-to:exploring:incoming-outgoing>`, can be used to access all neighbors from a certain node and provide advanced filtering options.
+However, often one doesn't need this expressivity and simply wants to retrieve all neighboring nodes with a syntax that is as succint as possible.
+A prime example is to retrieve the *inputs* or *outputs* of :ref:`a process <topics:processes:concepts>`.
+Instead of using :meth:`~aiida.orm.nodes.node.Node.get_incoming` and :meth:`~aiida.orm.nodes.node.Node.get_outgoing`, to get the inputs and outputs of a ``process_node`` one can do:
+
+.. code-block:: python
+
+    inputs = process_node.inputs
+    outputs = process_node.outputs
+
+These properties do not return the actual inputs and outputs directly, but instead return an instance of :class:`~aiida.orm.utils.managers.NodeLinksManager`
+The reason is because through the manager, the inputs or outputs are accessible through their link label and it is tab-completed.
+For example, if the ``process_node`` has an output with the label ``result``, it can be retrieved as:
+
+.. code-block:: python
+
+    process_node.outputs.result
+
+The inputs or outputs can also be accessed through key dereferencing:
+
+.. code-block:: python
+
+    process_node.outputs['result']
+
+If there is no neighboring output with the given link label, a :class:`~aiida.common.exceptions.NotExistentAttributeError` or :class:`~aiida.common.exceptions.NotExistentKeyError` will be raised, respectively.
+
+.. note::
+
+    The ``inputs`` and ``outputs`` properties are only defined for :class:`~aiida.orm.nodes.process.process.ProcessNode`'s.
+    This means that you cannot *chain* these calls, because an input or output of a process node is guaranteed to be a :class:`~aiida.orm.nodes.data.Data` node, which does not have inputs or outputs.
```

Comment:
I would put this comment below, after explaining creator, to say that inputs+creator allows to go up in the provenance graph as much as you want, chaining them. Instead, going down, you cannot for the reason you mention

Reply:
Not sure where exactly you intend this to be placed. It is true that chaining `creator` and `inputs` alternatively works, but that is just one example and not every node has a creator. Instead, I kept this as is and added another note after the explanation of `creator` that gives the example you propose.